gymnasium/wrappers/record_episode_statistics.py

```diff
@@ -114,38 +85,30 @@ def step(self, action):
         ), f"`info` dtype is {type(infos)} while supported dtype is `dict`. This may be due to usage of other wrappers in the wrong order."
         self.episode_returns += rewards
         self.episode_lengths += 1
-        if not self.is_vector_env:
-            terminateds = [terminateds]
-            truncateds = [truncateds]
-        terminateds = list(terminateds)
-        truncateds = list(truncateds)
-
-        for i in range(len(terminateds)):
-            if terminateds[i] or truncateds[i]:
-                episode_return = self.episode_returns[i]
-                episode_length = self.episode_lengths[i]
-                episode_info = {
-                    "episode": {
-                        "r": episode_return,
-                        "l": episode_length,
-                        "t": round(time.perf_counter() - self.t0, 6),
-                    }
-                }
-                if self.is_vector_env:
-                    infos = add_vector_episode_statistics(
-                        infos, episode_info["episode"], self.num_envs, i
-                    )
-                else:
-                    infos = {**infos, **episode_info}
-                self.return_queue.append(episode_return)
-                self.length_queue.append(episode_length)
-                self.episode_count += 1
-                self.episode_returns[i] = 0
-                self.episode_lengths[i] = 0
+        dones = truncateds | terminateds
+        num_dones = np.sum(dones)
+        if num_dones:
+            episode_info = {
+                "episode": {
+                    "r": np.where(dones, self.episode_returns, 0),
+                    "l": np.where(dones, self.episode_lengths, 0),
+                    "t": np.full(
+                        (self.num_envs,), round(time.perf_counter() - self.t0, 6)
```

Comment:
I'm not sure this will matter much but should be use `np.round` rather than the python `round` function given the value gets passed directly to a numpy array.

Secondly, this "t" value doesn't seem much help as `t0` is never updated to my eyes. So "t" seems to be the number of milli (or nano) seconds since the wrapper was initialised. 

This would break backward compatibility but would we want to change "t" to be the number of milli (or nano) seconds that the episode has taken. This would require us to have a `perf_counter` for every sub-environment but this shouldn't be too bad